### PR TITLE
WT-8318 Add format.sh support for a directory of CONFIG files

### PR DESCRIPTION
**Pull request by keithbostic**
Add the -D option to read a directory for CONFIG files.
Fix a bug where -c with an absolute path name argument would fail.

### CODE REVIEW

File: test/format/format.sh [hunk 1]
```diff
@@ -94,6 +100,18 @@ while :; do
 	-c)
 		config="$2"
 		shift ; shift ;;
+	-D)
+		[[ "$2" == /* ]] || {
+		    echo "$name: -D argument must be an absolute path"
```
Comment (lukech) [inline test/format/format.sh:105]: Is it necessary to only allow an absolute path?

File: test/format/format.sh [hunk 2]
```diff
@@ -94,6 +100,18 @@ while :; do
...
+		    exit 1
+		}
+		directory_list=($2/CONFIG.*)
+		directory_total=${#directory_list[@]}
+		[[ -f "${directory_list[0]}" ]] || {
+		    echo "$name: no CONFIG files found in $2"
+		    exit 1
+		}
```
Comment (lukech) [inline test/format/format.sh:113]: Assume this section of code is to check if the given directory is empty or not, there are a few [more elegant ways](https://superuser.com/questions/352289/bash-scripting-test-for-empty-directory) of achieving it, e.g.
```
#!/bin/bash
if [ -d /path/to/dir ]; then
    # the directory exists
    [ "$(ls -A /path/to/dir)" ] && echo "Not Empty" || echo "Empty"
else
    # You could check here if /path/to/dir is a file with [ -f /path/to/dir]
fi
```

Comment (keithbostic) [reply test/format/format.sh:113]: @lukech, this check isn't just for an empty directory, it's checking if we found a file in the directory matching the `CONFIG.XXX` expansion (which should includes non-existent and empty directories as well). Is there a better way to do all that?

File: test/format/format.sh (hunk 1 above)
Comment (keithbostic) [reply test/format/format.sh]: No; fixed in 2da3291ff.

File: test/format/format.sh (hunk 2 above)
Comment (lukech) [reply test/format/format.sh]: Right. I'm fine with the current implementation.

**Review by lukech — APPROVED** [review]
LGTM

**Pull request merged**